prow/tide/tide.go

```diff
@@ -123,13 +123,82 @@ func pickSmallestPassingNumber(prs []pullRequest) (bool, pullRequest) {
 	return smallestNumber > -1, smallestPR
 }
 
+// accumulateBatch returns a list of PRs that can be merged after passing batch
+// testing, if any exist. It also returns whether or not a batch is currently
+// running.
+func accumulateBatch(presubmits []string, prs []pullRequest, pjs []kube.ProwJob) ([]pullRequest, bool) {
+	prNums := make(map[int]pullRequest)
+	for _, pr := range prs {
+		prNums[int(pr.Number)] = pr
+	}
+	type accState struct {
+		prs       []pullRequest
+		jobStates map[string]simpleState
+		// Are the pull requests in the ref still acceptable?
+		validPulls bool
+	}
+	states := make(map[string]*accState)
+	for _, pj := range pjs {
+		if pj.Spec.Type != kube.BatchJob {
+			continue
+		}
+		// If any batch job is pending, return now.
+		if toSimpleState(pj.Status.State) == pendingState {
+			return nil, true
+		}
+		// Otherwise, accumulate results.
+		ref := pj.Spec.Refs.String()
+		if _, ok := states[ref]; !ok {
+			states[ref] = &accState{
+				jobStates:  make(map[string]simpleState),
+				validPulls: true,
+			}
+			for _, pull := range pj.Spec.Refs.Pulls {
+				if pr, ok := prNums[pull.Number]; ok && string(pr.HeadRef.Target.OID) == pull.SHA {
+					states[ref].prs = append(states[ref].prs, pr)
+				} else {
+					states[ref].validPulls = false
+					break
+				}
+			}
+		}
+		if !states[ref].validPulls {
```

Comment:
Add a comment explaining the reasons why a batch is not valid anymore.

Reply:
Done.